linera-core/src/chain_worker/state/mod.rs

```diff
@@ -353,21 +353,33 @@ where
         let pending_blobs = &self.chain.manager.get().pending_blobs;
 
         let mut found_blobs = Vec::new();
+        let mut missing_indices = Vec::new();
         let mut missing_blob_ids = Vec::new();
-        for blob_id in blob_ids {
+        for (index, blob_id) in blob_ids.iter().enumerate() {
             if let Some(blob) = pending_blobs.get(blob_id) {
-                found_blobs.push(blob.clone());
-            } else if let Ok(blob) = self.storage.read_blob(*blob_id).await {
-                found_blobs.push(blob);
+                found_blobs.push(Some(blob.clone()));
             } else {
+                found_blobs.push(None);
+                missing_indices.push(index);
                 missing_blob_ids.push(*blob_id);
```

Comment:
The `blob_ids` argument is a `HashSet` anyway, so the order doesn't matter. I think it's enough to collect the `missing_blob_ids` here and not deal with indices. Below we can then try to read the missing blobs from storage and if any of them are `None`, we return `BlobsNotFound`.

Reply:
I agree, the indices here seem unnecessary complexity

Reply:
Ok, I changed accordingly.